scanner/cmd/scannerctl/main.go

```diff
@@ -30,20 +32,34 @@ func main() {
 		utils.CrashOnError(os.Setenv(mtls.KeyFileEnvName, filepath.Join(*certsPath, mtls.ServiceKeyFileName)))
 	}
 
-	if len(flag.Args()) < 1 {
-		log.Fatalf("Missing <image-url>")
+	tlsOpts := clientconn.TLSConfigOptions{
+		UseClientCert: clientconn.MustUseClientCert,
 	}
 
-	imageURL := flag.Args()[0]
-	ctx := context.Background()
-	tlsConfig, err := clientconn.TLSConfig(mtls.ScannerSubject, clientconn.TLSConfigOptions{
-		UseClientCert: clientconn.MustUseClientCert,
-	})
+	var dialTarget string
+	if *serverName != "" {
+		tlsOpts.ServerName = *serverName
+		dialTarget = fmt.Sprintf("%s:%d", *serverName, *port)
+	} else {
+		dialTarget = fmt.Sprintf(":%d", *port)
+	}
+
+	if ca, err := mtls.LoadDefaultCA(); err == nil {
+		tlsOpts.RootCAs = ca.CertPool()
+	}
```

Comment:
Can you clarify why just setting the variables is not enough and we need the additional `tlsOpts`?

Reply:
We need a signature from a root CA we trust to trust the server certificate we're getting back. Without this option, we'll receive the following error:
```sh
❯ go run cmd/scannerctl/main.go \
    -certs certs/client \
    -server-name localhost \
    -port 8443 \
    docker.io/library/hello-world:latest
2023/10/03 11:02:31 Reply: <nil> (rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509:
“server.scanner.stackrox” certificate is not trusted")
```

There might be a way to sign the certs with `{mtls.ScannerSubject}.stackrox`, which the code looks for by default, but I'm unsure how to do that. That being said, `roxctl` [does something similar](https://github.com/stackrox/stackrox/blob/5fa874c77c586d191148b8fc98c59343a1841a5c/roxctl/common/tls.go#L77-L114) to add the root CA (assuming my understanding of what's going on in the `roxctl` code is correct)